programs/ziti-edge-tunnel/netif_driver/linux/tun.c

```diff
@@ -24,8 +29,25 @@
 #define IP_BIN "/sbin/ip "
 #endif
 
+#define CHECK_UV(op) do{ int rc = op; if (rc < 0) ZITI_LOG(ERROR, "uv_err: %d/%s", rc, uv_strerror(rc)); } while(0)
+
 #define RESOLVECTL "resolvectl"
 
+static struct {
+    char tun_name[16];
+    uint32_t dns_ip;
+
+    uv_udp_t nl_udp;
+    uv_timer_t update_timer;
+} dns_maintainer;
+
+typedef struct nl_req_s nl_req_t;
```

Comment:
Is there a reason you didn't combine the typedef with the struct definition?

Reply:
this is not needed (yet?), cleaning it up